cpp/tests/io/json_tree.cpp

```diff
@@ -0,0 +1,572 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <io/json/nested_json.hpp>
+#include <io/utilities/hostdevice_vector.hpp>
+
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+#include <string>
+
+namespace cuio_json = cudf::io::json;
+
+namespace cudf::io::json {
+// Host copy of tree_meta_t
+struct tree_meta_t2 {
+  std::vector<NodeT> node_categories;
+  std::vector<NodeIndexT> parent_node_ids;
+  std::vector<TreeDepthT> node_levels;
+  std::vector<SymbolOffsetT> node_range_begin;
+  std::vector<SymbolOffsetT> node_range_end;
+};
+}  // namespace cudf::io::json
+
+namespace {
+std::string get_node_string(std::size_t const node_id,
+                            cuio_json::tree_meta_t2 const& tree_rep,
+                            std::string const& json_input)
+{
+  auto node_to_str = [] __host__ __device__(cuio_json::PdaTokenT const token) {
```

Comment:
```suggestion
  auto node_to_str = [] (cuio_json::PdaTokenT const token) {
```
This does not need `__device__` decl.